src/NServiceBus.Core/Pipeline/Outgoing/OutgoingContext.cs

```diff
@@ -11,10 +11,11 @@ protected OutgoingContext(string messageId, Dictionary<string, string> headers,
             : base(parentContext)
         {
             Headers = headers;
-            MessageId = messageId;
+
+            Headers[NServiceBus.Headers.MessageId] = messageId;
```

Comment:
wouldn't this mean that we replace the header every time we create a new subclass of `OutgoingContext` (e.g. `RoutingContext`) which may override the configured header in a mutator too?